src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,161 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import java.io.*;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.PrintWriter;
+
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+
+public class MovieRecommender{
+	public static void main(String args[]){
+
+	}
+
+	public Map<String,Integer> mapProducts = new HashMap<String,Integer>();
+	public Map<String,Integer> mapUsers = new HashMap<String,Integer>();
+	public int totalReviews = 0;
+	public int totalProducts = 0;
+	public int totalUsers = 0;
+
+	public MovieRecommender(String path){
+		toCSV(path);
+	}
+
+	public void toCSV(String path){
+		String encoding = "UTF-8";
+		InputStream fileStream;
+		InputStream gzipStream;
+		Reader decoder;
+		BufferedReader buffered = null;
+		ArrayList<String> products = new ArrayList<String>();
+		ArrayList<String> users = new ArrayList<String>();
+		ArrayList<Double> scores = new ArrayList<Double>();
+
+		try{
+			fileStream = new FileInputStream(path);
+			gzipStream = new GZIPInputStream(fileStream);
+			decoder = new InputStreamReader(gzipStream, encoding);
+			buffered = new BufferedReader(decoder);
+		}catch(IOException e){}
+
+
+		String s="";
+		try{
+			while((s=buffered.readLine())!=null){
+				//System.out.println(s);
+				if(s.contains("product/productId: ")){
+					String[] arr = s.split("product/productId: ");
+					//System.out.println(arr[1]);
+					products.add(arr[1]);
+				}else if(s.contains("review/userId: ")){
+					String[] arr = s.split("review/userId: ");
+					//System.out.println(arr[1]);
+					users.add(arr[1]);
+				}else if(s.contains("review/score: ")){
+					String[] arr = s.split("review/score: ");
+					//System.out.println(arr[1]);
+					scores.add(Double.parseDouble(arr[1]));
+					this.totalReviews++;
+				}
+			}
+		}catch(Exception e){}
```

Comment:
You are not doing anything with the exception, you may log the error or propagate it :) it's important to consider what to do if the exception occurs :)